docs/reference/settings/common-defs.asciidoc

```diff
@@ -0,0 +1,167 @@
+tag::ssl-certificate[]
+Specifies the path for the PEM encoded certificate (or certificate chain) that is
+associated with the key.
+//TBD: This setting can be used only if `ssl.key` is set.
+end::ssl-certificate[]
+
+tag::ssl-certificate-authorities[]
+List of paths to PEM encoded certificate files that should be trusted.
+//TBD: You cannot use this setting and `ssl.truststore.path` at the same time.
+end::ssl-certificate-authorities[]
+
+tag::ssl-cipher-suites-values[]
+Supported cipher suites can be found in Oracle's
+https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-7093246A-31A3-4304-AC5F-5FB6400405E2[Java Cryptography Architecture documentation].
+The default is `TLS_AES_256_GCM_SHA384`, `TLS_AES_128_GCM_SHA256`,
+`TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384`, `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256`,
+`TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384`, `TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256`,
+`TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384`, `TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256`,
+`TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384`, `TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256`,
+`TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA`, `TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA`,
+`TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA`, `TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA`,
+`TLS_RSA_WITH_AES_256_GCM_SHA384`, `TLS_RSA_WITH_AES_128_GCM_SHA256`,
+`TLS_RSA_WITH_AES_256_CBC_SHA256`, `TLS_RSA_WITH_AES_128_CBC_SHA256`,
+`TLS_RSA_WITH_AES_256_CBC_SHA`, `TLS_RSA_WITH_AES_128_CBC_SHA`.
+end::ssl-cipher-suites-values[]
+//TBD: Are these two different definitions (with different Oracle URLs) for cipher_suites required?
```

Comment:
On master, no because we don't support Java 8 any more.
On 7.x, I think so. The set of options (and defaults) are different depend in the Java version.

Although, strictly speaking we have different values between Java11 and Java12 as well (since #42155) so we may need to think this through a bit more.

@jkakavas What do we want to do with FIPS here? BC JSSE doesn't support TLS1.3, so some of these ciphers aren't supported, but I think technically the default is still correct (we don't change the default cipher suite list based on the supported protocols list).

Reply:
Yes the default  is still that one so I agree we shouldn't add anything here (We could add something in the FIPS docs though, but this is part of the effort to make FIPS docs better ).
 
The actual list of unusable ones while running with fips mode ( we do print this in the logs on startup too)
JDK11: 
```
TLS_AES_256_GCM_SHA384
TLS_AES_128_GCM_SHA256
TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
TLS_RSA_WITH_AES_256_GCM_SHA384
TLS_RSA_WITH_AES_128_GCM_SHA256
```

JDK13:
```
TLS_AES_256_GCM_SHA384
TLS_AES_128_GCM_SHA256
TLS_CHACHA20_POLY1305_SHA256
TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
TLS_RSA_WITH_AES_256_GCM_SHA384
TLS_RSA_WITH_AES_128_GCM_SHA256
```

Reply:
Thanks, @jkakavas I've drafted those changes in https://github.com/elastic/elasticsearch/pull/53138

Reply:
> The set of options (and defaults) are different depend in the Java version. Although, strictly speaking we have different values between Java11 and Java12 as well (since #42155) so we may need to think this through a bit more.

@tvernum For now I have (1) deleted the "ssl-cipher-suites-values-java8" definition, (2) moved the Java11 info into a "ssl-cipher-suites-values-java11" definition, (3) put a Java12 URL in the "ssl-cipher-suites-values" definition and added the three ChaCha20 ciphers, (4) updated all occurrences of *-values-java8 and *-values-java11 to use the generic *-values.  I'm just keeping the *-java11 for now in case we backport this PR.

Reply:
Thanks @lcawl that sounds right to me.